events-api/src/event_traits.rs

```diff
@@ -35,7 +38,32 @@ impl EventSource {
     /// New event source with default values.
     pub fn new(node: String) -> Self {
         let component = COMPONENT.get().cloned().unwrap_or_default().into();
-        Self { component, node }
+        Self {
+            component,
+            node,
+            ..Default::default()
+        }
+    }
+
+    /// Add rebuild event specific data to event source.
+    pub fn add_rebuild_data(
+        self,
+        status: RebuildStatus, // Rebuild status
+        source: Option<&str>,  // Rebuild source replica uri
```

Comment:
why is source optional?

Reply:
Made it optional for the rebuild events where I couldn't find the source details.

Reply:
Could you point me to such a place?

Reply:
Removed optional as the source_uri can be found for all the rebuild events.